### PR TITLE
CashuWallet: update receive function

### PR DESCRIPTION
**Pull request by BilligsterUser**

### CODE REVIEW

**Comment by BilligsterUser** [comment]
Review pls @gandlafbtc

File: src/CashuWallet.ts [hunk 1]
```diff
@@ -218,32 +243,32 @@ class CashuWallet {
...
+	private createBlankOutputs(feeReserve: number): BlindedMessageData {
 		const blindedMessages: Array<SerializedBlindedMessage> = [];
 		const secrets: Array<Uint8Array> = [];
 		const rs: Array<bigint> = [];
 		const count = Math.ceil(Math.log2(feeReserve));
 		for (let i = 0; i < count; i++) {
 			const secret = randomBytes(32);
 			secrets.push(secret);
-			const { B_, r } = await dhke.blindMessage(secret);
+			const { B_, r } = dhke.blindMessage(secret);
```
Comment (gandlafbtc) [inline src/CashuWallet.ts:271]: did this change because we use the new curve lib? or how come this is no longer async

File: src/utils.ts [hunk 2]
```diff
@@ -72,14 +72,43 @@ function handleTokens(token: string): Token {
...
+/**
+ * merge proofs from same mint,
+ * removes TokenEntrys with no proofs or no mint field
+ * and sorts proofs by id
+ *
+ * @export
+ * @param {Token} token
+ * @return {*}  {Token}
+ */
+export function cleanToken(token: Token): Token {
```
Comment (gandlafbtc) [inline src/utils.ts:92]: would be great to have a test for that funtion!

File: src/CashuWallet.ts [hunk 3]
```diff
@@ -177,24 +168,58 @@ class CashuWallet {
...
+				amount2BlindedMessages.secrets,
+				keys
+			);
+			proofs.push(...proofs1, ...proofs2);
+		} catch (error) {
+			console.error(error);
+			proofsWithError.push(...tokenEntry.proofs);
+		}
+		return {
+			proofs,
```
Comment (gandlafbtc) [inline src/CashuWallet.ts:206]: I'm not even sure if we need this. Can there be single proofs that fail in a token, while others succeed?

**Review by gandlafbtc — COMMENTED** [review]
Amazing PR :fire:

File: src/CashuWallet.ts (hunk 1 above)
Comment (BilligsterUser) [reply src/CashuWallet.ts:271]: Yes

File: src/CashuWallet.ts (hunk 3 above)
Comment (gandlafbtc) [reply src/CashuWallet.ts:206]: oh nevermind.

Comment (gandlafbtc) [reply src/CashuWallet.ts:206]: sorry for the confusion, i see now that you need them in the receive function to determine wether the token was faulty. all good!

File: src/utils.ts (hunk 2 above)
Comment (BilligsterUser) [reply src/utils.ts:92]: done

**Comment by BilligsterUser** [comment]
after this is merged i will work on checking for changed keys before creating proofs

File: src/utils.ts (hunk 2 above)
Comment (BilligsterUser) [reply src/utils.ts:92]: thank you for reviewing my prs ❤️

**Pull request merged**